### ISSUE TITLE
It gives me error on Linux Mint 17.2

### ISSUE THREAD

**Issue by coworkingon-es** [projecthamster/cinnamon-applet#20]
<----------------
@/home/maria/.local/share/cinnamon/applets/hamster@projecthamster.wordpress.com/applet.js:22
Extension.prototype._init@/usr/share/cinnamon/js/ui/extension.js:180
Extension@/usr/share/cinnamon/js/ui/extension.js:138
loadExtension@/usr/share/cinnamon/js/ui/extension.js:468
onEnabledAppletsChanged@/usr/share/cinnamon/js/ui/appletManager.js:207
---------------->
error t=2015-09-29T13:41:00.334Z [Applet "hamster@projecthamster.wordpress.com"]: Error importing applet.js from hamster@projecthamster.wordpress.com
error t=2015-09-29T13:41:00.335Z Could not load applet hamster@projecthamster.wordpress.com

**Comment by jonbrett** [comment]
Similar error was reported on Mint 17.1 (#14). Try following the steps there and let me know if they work for you.

**Comment by sietse** [comment]
### Preamble

Alright.
I, too, tried to get the Hamster applet to work on Cinnamon 2.6.13 in Linux Mint 17.2, and found it didn't work out of the box.
I have just now gotten it to work, and below are what I think were the relevant steps.
Because I tried out a great many things, I've had to guess at what got it to work in the end. 
Therefore, the instructions may be unreliable.
Sorry.

If the instructions don't work for you, give me a shout (reply to this comment) and I'll see if I have time to set up a vagrantbox and figure it out in a pristine environment, but for now I'm going to leave them like this.

### In brief

I uninstalled everything hamster-related, including the Cinnamon applet.
I installed from the git repo, branch `hamster_stable` (commit `a5bda5a59abf` dated 2015-09-11).
That got me a working Hamster.
I thought that also got me a working applet, but it turns out to be just a system tray I did not need to install `hamster/cinnamon-applet`.
But I have already written this entire post, and part of getting the Cinnamon applet to work is getting Hamster itself to work, so I'm posting this anyway.

### Instructions wot I think might work

1. Uninstall the hamster-applet that the package manager provided:

  ```
  sudo aptitude uninstall hamster-applet hamster-indicator
  ```

  It may not uninstall correctly: at one point I discovered a wrong version of some hamster python files (`storage.Storage` was giving problems), and I presume they were left there during of my many installs+uninstalls this evening.
  So to be on the safe side, run this:

  ```
  cd /usr/local/lib/python2.7/dist-packages/
  sudo rm -rf hamster
  ```

1. Also uninstall Cinnamon's hamster panel applet, if you already installed it:

  Right-click on panel > Add applets to the panel > search for hamster > [right click] > Uninstall

1. Kill any hamster processes that may be still running:

  ```
  killall --regex --interactive '*hamster*'
  ```

1. Clone the Hamster time tracker repo, and update your repo to the 'hamster-stable' branch.
  I'm using Mercurial plus
  [hg-git](https://hg-git.github.io/), and I don't feel like hiding it, so have some Mercurial commands with the git stuff in comments.
  Feel free to use git yourself. Just know that you don't have to. ;-)

  ```
  # git clone https://github.com/projecthamster/hamster 
  hg clone git+https://github.com/projecthamster/hamster 

  cd hamster

  # git checkout hamster-stable
  hg update hamster-stable
  ```

  (Reason for using Mercurial even on git repos: because I'm in love with querying history with user-friendly
  [revets](https://www.selenic.com/hg/help/revsets)
  and formatting output with user-friendly
  [templates](https://www.selenic.com/hg/help/templates)
  and generally having user-friendly
  [non-insane](http://stevelosh.com/blog/2013/04/git-koans/)
  commands.)

1. Now follow the instructions in the repo's `README.textile` to configure Hamster.

  ```
  ./waf configure
  ```

  This also checks whether all required packages are available. Things I found myself having to install (there may have been more, I can't remember):

  ```
  sudo aptitude install gnome-doc-utils python-wnck
  ```

  Don't bother installing docbook2xml, I couldn't find it and it runs fine without it.
  For the rest, just keep running `./waf configure`, googling the missing bits, and installing them.
  Repeat until you have everything (except docbook2xml).

1. Once `./waf configure` goes all green, build and install Hamster.

   ```
   ./waf build --prefix=/usr/local
   sudo ./waf install
   ```

1. Now hamster should work.
   Try running `hamster` from the command line.
   I got a nice window *and* a panel applet, both at the same time.
   I hope you do, too!


### Notes and observations

* I tried building and installing the `master` branch of Hamster (`8c739feccb9`, dated 2015-11-04), but the resulting window seemd to be missing controls.

* At some point I ran

  ```
  cd /usr/share/glib-2.0/schemas
  glib-compile-schemas .
  ```

  That's from [Issue #18](https://github.com/projecthamster/cinnamon-applet/issues/18)
  I don't think it was necessary, in the end, This is why it's not in the instructions. But it may have helped, so I'm mentioning it here.

* And *now*, at the end of this monster post, I discover that what I thought was an applet not an applet so much as a 'Hamster is running' system tray icon.
   So I still get to try to get the Cinnamon applet working.
   Joy.
   Sorry.

* Unless I'm very wrong, I'm only seeing 'info'-level warnings in Looking Glass.
  Didn't see a single warning or error or trace, even while the hamster cinnamon panel applet was failing to activate. Tried googling this problem, got nowhere.

In conclusion: Sorry about this braindump, and sorry that I only found out once I'd finished it that I had gotten the wrong applet working. :-/ Still, I think getting the Hamster program itself to run is a good start.

**Comment by plutocrat** [comment]
Hi,
Also having problems with this on Cinnamon / Mint 17.3
When I try to add the app I get
```
error t=2016-07-29T04:18:30.422Z Schema "org.cinnamon.hamster-applet" not found.
error t=2016-07-29T04:18:30.422Z [Applet "hamster@projecthamster.wordpress.com"]: Failed to evaluate 'main' function on applet: hamster@projecthamster.wordpress.com/29
error t=2016-07-29T04:18:30.422Z Could not load applet hamster@projecthamster.wordpress.com
```
I've followed the very detailed setup instructions above using ./waf configure until all the dependencies are satisfied. (Needed the following if that helps anyone avoid some googling. 
sudo apt-get install libglib2.0-dev libdbus-glib-1-dev docbook-utils gnome-doc-utils xsltproc gnome-doc-utils xsltproc )
Used /waf build --prefix=/usr rather than /usr/local, but that shouldn't affect anything should it? 
Hamster seems to run OK, but I don't have much control over starting and stopping tasks without the cinnamon applet.

**Comment by border0464111** [comment]
@coworkingon-es did not give any news regarding this issue. 

For @plutotcrat "Failed to evaluate 'main' function on applet: hamster@projecthamster.wordpress.com/29"

We now know it's was due to shema rebuild issue who should be solved now.

**Comment by plutocrat** [comment]
Thanks. I actually struggled with this version of Hamster, as it seemed to have removed several features from the interface. I'm currently running Linux Mint 18.1 with the version of hamster supplied in the standard Ubuntu repositories, which does what I need.

**Comment by elbenfreund** [comment]
As this [Issue](https://github.com/projecthamster/cinnamon-applet/issues/30) documents the cinnamon applet is now maintained by the Mint Linux team.
As such this repository is kept around for historical and migration reasons.
Contributors are encouraged to head over there and no work will be done on this
repository at all.
Thanks for your support.
